app/app.go

```diff
@@ -600,6 +600,7 @@ func NewTgradeApp(
 		if err := app.twasmKeeper.InitializePinnedCodes(ctx); err != nil {
 			tmos.Exit(fmt.Sprintf("failed initialize pinned codes %s", err))
 		}
+		app.poeKeeper.InitContractAddressCache(ctx)
```

Comment:
Looks good.

I imagine this is affecting the order of writes to the store between contract generated stuff, and modules (bank, etc.) generated stuff.